Libraries/LibWeb/CredentialManagement/PasswordCredential.h

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (c) 2025, Altomani Gianluca <altomanigianluca@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <LibJS/Forward.h>
+#include <LibWeb/Bindings/PasswordCredentialPrototype.h>
+#include <LibWeb/Bindings/PlatformObject.h>
+#include <LibWeb/CredentialManagement/Credential.h>
+#include <LibWeb/HTML/HTMLFormElement.h>
+
+namespace Web::CredentialManagement {
+
+class PasswordCredential final : public Credential {
+    WEB_PLATFORM_OBJECT(PasswordCredential, Credential);
+    GC_DECLARE_ALLOCATOR(PasswordCredential);
+
+public:
+    [[nodiscard]] static GC::Ref<PasswordCredential> create(JS::Realm&);
+    static WebIDL::ExceptionOr<GC::Ref<PasswordCredential>> construct_impl(JS::Realm&, HTML::HTMLFormElement const&);
+    static WebIDL::ExceptionOr<GC::Ref<PasswordCredential>> construct_impl(JS::Realm&, PasswordCredentialData const&);
+
+    virtual ~PasswordCredential() override;
+
+    String const& password() { return m_password; }
+
+    String type() override { return "password"_string; }
+
+private:
+    explicit PasswordCredential(JS::Realm&);
+    virtual void initialize(JS::Realm&) override;
+
+    String m_password;
```

Comment:
I'd prefer using Core::SecretString here, to ensure the key material is erased when the object is destroyed. Though it looks like we yeeted that class. it should be restored from git in a follow-up PR then.

Reply:
Added a TODO for it